src/components/SectionNotification/SectionNotification.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { TPAComponentProps } from '../../types';
+import { TPAComponentsConsumer } from '../TPAComponentsConfig';
+import { SECTION_NOTIFICATION_DATA_HOOKS } from './dataHooks';
+import styles from './SectionNotification.st.css';
+import {
+  NOTIFICATION_TYPE,
+  SectionNotificationDefaultProps,
+  SectionNotificationProps,
+} from './types';
+
+/** Section notification displays an important, succinct message, and provides actions for users to address and can not be dismissed.  */
+export class SectionNotification extends React.Component<
+  SectionNotificationProps & TPAComponentProps
+> {
+  static displayName = 'SectionNotification';
+  static defaultProps: SectionNotificationDefaultProps = {
+    type: NOTIFICATION_TYPE.default,
+    'data-hook': SECTION_NOTIFICATION_DATA_HOOKS.root,
+  };
+
+  render() {
+    const { icon, text, controls, type } = this.props;
+    const isError = type === NOTIFICATION_TYPE.error;
+
+    return (
+      <TPAComponentsConsumer>
+        {({ mobile }) => (
+          <div
+            {...styles('root', {
+              error: isError,
+            })}
+            data-error={isError}
+            data-mobile={mobile}
+            data-hook={this.props['data-hook']}
+            aria-live="assertive"
+          >
+            <div className={styles.main}>
+              <div className={styles.textWrapper}>
+                <div
+                  className={styles.icon}
+                  data-hook={SECTION_NOTIFICATION_DATA_HOOKS.iconWrapper}
+                >
+                  {icon}
+                </div>
+                <div
+                  className={styles.text}
+                  data-hook={SECTION_NOTIFICATION_DATA_HOOKS.textWrapper}
+                >
+                  {text}
+                </div>
+              </div>
+              {!isError && controls ? (
+                <div
+                  className={styles.controls}
+                  data-hook={SECTION_NOTIFICATION_DATA_HOOKS.controlsWrapper}
+                >
+                  {controls}
+                </div>
+              ) : null}
+            </div>
+          </div>
+        )}
+      </TPAComponentsConsumer>
```

Comment:
So there's a bit of a change I think we should make here.
It might seem big, but it shouldn't be too problematic. I hope 🙏 

First, from looking at the `st.css` file, I see that you used `flex-wrap` which is great.
This means we don't really need a special case for `mobile`, so we can remove the `TPAComponentsConsumer` and the `mobile` state.

Second, we need more control over the content that's inserted to this component.
What I suggest is to create a compound component, so that consumers will be able to use this component like this:
```jsx
<SectionNotification>
  <SectionNotification.Icon icon={<ErrorIcon/>} />
  <SectionNotification.Text>
    Lorem ipsum...
  </SectionNotification.Text>
  <SectionNotification.Button priority={PRIORITY.primary} onClick={() => {}}>
    Approve
  </SectionNotification.Button>
  <SectionNotification.Button priority={PRIORITY.secondary} onClick={() => {}}>
    Decline
  </SectionNotification.Button>
</SectionNotification>
```

So what this means is just moving the HTML you wrote, and move it to a "child" component:
```jsx
function NotificationText (props) {
  return (
    <div className={styles.text} 
         data-hook={SECTION_NOTIFICATION_DATA_HOOKS.textWrapper}>
      {props.children}
    </div>
  );
}
```
etc.

Regarding the button, it should render our Button component, and just pass the prop that it receives, so that consumers will have enough flexibility.
In addition, the button should also have an additional prop: `type`, with the values: `default` and `text`.
`default` should render a regular Button component, and the `text` should render a `TextButton` component.

All this let's us have control on what goes in the component, and it's styling.
For instance, the `SectionNotification.Button`'s will automatically have the margins between them as in the spec.